resources/functions/user-management/index.py

```diff
@@ -1,98 +1,141 @@
+    logger.info(users.serialize())
+    return_response = {
+        "data": users.serialize()
+    }
+
+    if next_token:
+        return_response["next_token"] = next_token
 
-@app.get("/users/<username>")
+    return return_response, HTTPStatus.OK
+
+
+@app.get("/users/<userId>")
 @tracer.capture_method
-def get_user(username):
-    user_details = {}
-    user_details['idpDetails'] = idp_details
-    user_details['userName'] = username
-    user = idp_user_mgmt_service.get_user(user_details)
-    if (user == None):
-        logger.info("User not found")        
-        return {"response": "User not found"}, HTTPStatus.NOT_FOUND.value
-    logger.info(user.serialize())    
-    return user.serialize(), HTTPStatus.OK.value
-
-@app.put("/users/<username>")
+def get_user(userId: Annotated[str, Path(min_length=0)]):
+    user = idp_user_mgmt_service.get_user({
+        'idpDetails': idp_details,
+        'userName': userId,
+    })
+    if user:
+        logger.info(user.serialize())
+        return {'data': user.serialize()}, HTTPStatus.OK
+    else:
+        logger.info("User not found")
+        raise NotFoundError(f"User {userId} not found.")
+
+
+@app.put("/users/<userId>")
 @tracer.capture_method
-def update_user(username):
-    user_details = app.current_event.json_body
-    user_details['idpDetails'] = idp_details
-    user_details['userName'] = username
+def update_user(userId: Annotated[str, Path(min_length=0)],
+                userRole: Annotated[Optional[str], Body(embed=True)],
+                email: Annotated[Optional[str], Body(embed=True)]) -> dict[str, str]:
+    user_details = {
+        'userName': userId,
+        'idpDetails': idp_details,
+    }
+    if userRole:
+        user_details['userRole'] = userRole
+    if email:
+        user_details['email'] = email
     response = idp_user_mgmt_service.update_user(user_details)
-    logger.info(response)    
-    return {"response": "User updated"}, HTTPStatus.OK.value
+    logger.info(response)
+    return {"message": "User updated"}, HTTPStatus.OK
 
-@app.delete("/users/<username>/disable")
+
+@app.delete("/users/<userId>/disable")
 @tracer.capture_method
-def disable_user(username):
+def disable_user(userId: Annotated[str, Path(min_length=0)]):
     user_details = {}
     user_details['idpDetails'] = idp_details
-    user_details['userName'] = username
+    user_details['userName'] = userId
     response = idp_user_mgmt_service.disable_user(user_details)
-    logger.info(response)    
-    return {"response": "User disabled"}, HTTPStatus.OK.value
+    logger.info(response)
+    return {"message": "User disabled"}, HTTPStatus.OK
+
 
-@app.put("/users/<username>/enable")
+@app.put("/users/<userId>/enable")
 @tracer.capture_method
-def enable_user(username):
+def enable_user(userId: Annotated[str, Path(min_length=0)]):
     user_details = {}
     user_details['idpDetails'] = idp_details
-    user_details['userName'] = username
+    user_details['userName'] = userId
```

Comment:
The use of `userId` provides a more unique way of identifying a user - especially with something like Auth0. So, using `userId` rather than `username` is a better approach. However, I don't think that the `username` claim for a user should be set to a `userId` value; `username` and `userId` are separate entities, and both are valid for a user 😎

Reply:
Good point. I definitely agree that `username` and `userId` are two separate entities.

The thing to consider here is that the Cognito API uses `username` for API operations. For example, if you look at the `enable_user(...)` function https://github.com/awslabs/sbt-aws/blob/f699f385c32d520c4651e01098e3bca704e02cbb/resources/functions/user-management/cognito_user_management_service.py#L133-L143 you can see that the username is used as part of the API call: https://github.com/awslabs/sbt-aws/blob/f699f385c32d520c4651e01098e3bca704e02cbb/resources/functions/user-management/cognito_user_management_service.py#L139

Docs for `admin_enable_user` API: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cognito-idp/client/admin_enable_user.html

So, in order to conform to the new Users API schema, what I'm doing here is using the `userId` (which end-users will pass in for lookups and other API calls) and mapping that internally to the `userName` so it can work with the Cognito API.

As I write this, one improvement that I think we need to make is to clarify to end-users what the value they should use for `userId`. This might require us to map the `userName` we get from a call such as `admin_get_user(...)` that we make as part of this function https://github.com/awslabs/sbt-aws/blob/f699f385c32d520c4651e01098e3bca704e02cbb/resources/functions/user-management/cognito_user_management_service.py#L74 to `userId` and pass that back to the user. The result would be that the end-user sees `userId` wherever they would typically see `userName` and so we avoid confusion. That's something we can iron out, though. Appreciate the comment!

Reply:
created issue: https://github.com/awslabs/sbt-aws/issues/63